### PR TITLE
Fix/city has the highest different in apartment rent

### PR DESCRIPTION
**Pull request by jalalcheff**

### CODE REVIEW

File: src/main/kotlin/Main.kt [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import dataSource.CsvDataSource
```
Comment (flutterdevtaha) [inline src/main/kotlin/Main.kt:1]: why did you need to change on main which is belong to Hasan work?

File: src/main/kotlin/interactor/GetCityHasTheHighestDifferentInApartmentRent.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    private val dataSource: CostOfLivingDataSource,
+)
+{
+
+    fun execute(limit: Int): CityEntity {
+        return if (limit==1)
+            dataSource.getAllCitiesData()
+                .filter(::excludeNullApartmentBedroom)
+                .sortedByDescending { it.findHighestDifferentInCitiesRent() }
+                .take(limit)
```
Comment (flutterdevtaha) [inline src/main/kotlin/interactor/GetCityHasTheHighestDifferentInApartmentRent.kt:15]: why did you add take after filter?

File: src/test/kotlin/interactor/GetCityHasTheHighestDifferentInApartmentRentTest.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        cities=dataSource.getAllCitiesData()
+
+    }
+
+    @Test
+    fun `should return city has highest different in apartment rent when given list of cities and limit equal to one`(){
+        // given limit
+        val limit=1
+        // when the city has highest different in apartment rent
+        val result = cityHasTheHighestDifferentInApartmentRent.execute(limit)
```
Comment (flutterdevtaha) [inline src/test/kotlin/interactor/GetCityHasTheHighestDifferentInApartmentRentTest.kt:28]: rename it to actual

File: src/test/kotlin/interactor/GetCityHasTheHighestDifferentInApartmentRentTest.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        val limit=-1
+        // when the limit is negative
+        val actual =Executable {cityHasTheHighestDifferentInApartmentRent.execute(limit)}
+        val expected=InvalidLimitException::class.java
+        // then
+        assertThrows(expected, actual)
+    }
+
+    @Test
+    fun `should throw exception when limit is greater than one `(){
```
Comment (flutterdevtaha) [inline src/test/kotlin/interactor/GetCityHasTheHighestDifferentInApartmentRentTest.kt:57]: there is a space in the last of the test case name

**Review by flutterdevtaha — CHANGES_REQUESTED** [review]
please do small changes, otherwise you are perfect

File: src/main/kotlin/Main.kt (hunk 1 above)
Comment (jalalcheff) [reply src/main/kotlin/Main.kt:1]: Hassan didn't add my function

File: src/main/kotlin/interactor/GetCityHasTheHighestDifferentInApartmentRent.kt (hunk 2 above)
Comment (jalalcheff) [reply src/main/kotlin/interactor/GetCityHasTheHighestDifferentInApartmentRent.kt:15]: Coz I want the highest different so I have to do filtering and sorting on function result then take the first one which is of course the highest one

**Comment by flutterdevtaha** [comment]
[Enhancement on Task 4](https://trello.com/c/8wnSZHLo/33-enhancement-on-task-4)
